packages/docs-utils/package.json

```diff
@@ -0,0 +1,30 @@
+{
+  "name": "@mui-internal/docs-utils",
```

Comment:
I see this was released 3 days ago https://www.npmjs.com/package/@mui-internal/docs-utils.

@michaldudak Could we:

1. Rename all of these to @mui/internal-*
2. Remove all the members in https://www.npmjs.com/settings/mui-internal/members. I don't have admin permissions I can't do it.
3. Isolate the @mui/internal-* packages to the Material UI, Joy UI, Base UI, MUI System npm packages. We need to publish the former for each commit, and the latter once a week. The two should be released with two different scripts.

Reply:
Hi, yes, that's the plan.

> Isolate the @mui/internal-* packages to the Material UI, Joy UI, Base UI, MUI System npm packages. 

As far as I know, `lerna version` does not support filtering, so having these packages in the same repo may force us to release on the same schedule as the public Core packages. We could write a custom script that resets their version after `lerna version` finishes, but it may be confusing for the person releasing the packages.
One solution I can think of would be moving these packages to a separate infra repository and setting up different release rules there. If we used conventional commits, we could automate the most (if not whole) part of the release process, including the changelog generation. What do you think?

cc @Janpot

Reply:
If the goal is to develop the docs package in conjunction with the core libraries then I think following the core packages release schedule is the only option. The docs package has a dependency on the core packages, if they live in the same monorepo, it will depend on an unreleased version of the core packages in the monorepo. Any release of the docs package that is separate from a release of the core packages will potentially depend on unreleased changes in core packages and be broken. This is not a new problem, we already run into this with the current setup. Just last month alone monorepo update in X was blocked at least twice for several days just because it had to wait for core packages to be released. We can develop infra in a separate repo, but it takes away the ability to develop core and docs together.

And so here's where the submodules approach potentially could benefit us, bringing core modules + docs module together in the X monorepo and you act as if X+core is one big pnpm workspaces.

Reply:
> Just last month alone monorepo update in X was blocked at least twice for several days just because it had to wait for core packages to be released

Oh, I didn't know this, but it makes sense, we are dogfooding new MUI System and Material UI APIs on the docs infra. 

> submodules approach

I was going to raise this, it's a new argument in the "pros" side.

But it feels like we could with some hacks release the infra packages more frequently than the product ones, and that the reliance on the release of the product packages is temporary, or that we could solve it with canary versions (under a different npm name to not clutter the npm version view).

---

For context,
- I landed here from https://github.com/mui/material-ui/issues/41260#issuecomment-1962640901.
- I'm pushing so we move to monthly releases for the product packages, with canary on the side.

Reply:
I agree, this could be an issue with docs infra packages as they rely heavily on Core projects. The code infra packages, however, don't (or they are about to be decoupled), and could be released any time.

Reply:
> But it feels like we could with some hacks release the infra packages more frequently than the product ones, and that the reliance on the release of the product packages is temporary, or that we could solve it with canary versions (under a different npm name to not clutter the npm version view).

This will be very tricky for X, because the X packages rely on published releases of core packages. If the docs rely on docs infra which relies on canary releases, then we end up with two versions of core packages in the docs.